Mathlib/CategoryTheory/Generator.lean

```diff
@@ -351,6 +351,30 @@ def IsDetector (G : C) : Prop :=
 def IsCodetector (G : C) : Prop :=
   IsCodetecting ({G} : Set C)
 
+
+section Equivalence
+
+theorem IsSeparator.isSeparator_obj_of_equivalence {G : C} (hC : IsSeparator G) (F : C ⥤ D)
```

Comment:
You could add the `IsSeparating` version above in the file:
```lean
lemma IsSeparating.of_equivalence
    {𝒢 : Set C} (h : IsSeparating 𝒢) {D : Type*} [Category D] (e : C ≌ D) :
    IsSeparating (e.functor.obj '' 𝒢) := fun X Y f g H ↦
  e.inverse.map_injective (h _ _ (fun Z hZ h ↦ by
    obtain ⟨h', rfl⟩ := (e.toAdjunction.homEquiv _ _).surjective h
    simp only [Adjunction.homEquiv_unit, Category.assoc, ← Functor.map_comp,
      H (e.functor.obj Z) (Set.mem_image_of_mem _ hZ) h']))
```
and then this one becomes:
```lean
lemma IsSeparator.of_equivalence
    {G : C} (h : IsSeparator G) {D : Type*} [Category D] (e : C ≌ D) :
    IsSeparator (e.functor.obj G) := by
  simpa using IsSeparating.of_equivalence h e
```